libs/docs/src/docs/2.1.0/helloworld-example-che.md

```diff
@@ -0,0 +1,372 @@
+                  protocol: http
+                  targetPort: 13131
+                - attributes:
+                    contributed-by: che-code.eclipse.org
+                    discoverable: false
+                    urlRewriteSupported: false
+                  exposure: public
+                  name: code-redirect-2
+                  protocol: http
+                  targetPort: 13132
+                - attributes:
+                    contributed-by: che-code.eclipse.org
+                    discoverable: false
+                    urlRewriteSupported: false
+                  exposure: public
+                  name: code-redirect-3
+                  protocol: http
+                  targetPort: 13133
+              env:
+                - name: CHE_DASHBOARD_URL
+                  value: '...'
+                - name: CHE_PLUGIN_REGISTRY_URL
+                  value: '...'
+                - name: CHE_PLUGIN_REGISTRY_INTERNAL_URL
+                  value: '...'
+                - name: OPENVSX_REGISTRY_URL
+                  value: '...'
+              image: 'quay.io/devfile/universal-developer-image:ubi8-38da5c2'
+              memoryLimit: 1024Mi
+              memoryRequest: 256Mi
+              sourceMapping: /projects
+              volumeMounts:
+                - name: checode
+                  path: /checode
+            name: universal-developer-image
+        ```
+
+        {% callout title="Note!" %}
+        Your devfile many differ depending on the version of Che being used.
+
+        Careful when editing and removing fields from the devfile for your workspace,
+        a portion of these initial fields are used to configure the workspace itself.
+        The following steps will show what changes can be made to meet our goal.
+        {% /callout %}
+
+5. Change the value following component fields to set the component name to `runtime`
+    - `name` is the identifier used to refer to the component
+    - `che-code.eclipse.org/contributed-container` is a Eclipse Che specific field for referring to the contributed container component
+    by the `name`
+
+        ```yaml {% title="Field changes for changing the component name" %}
+        components:
+          - attributes:
+              ...
+              che-code.eclipse.org/contributed-container: runtime
+            ...
+            name: runtime
+        ```
+
+6. Change the `sourceMapping` to `/projects/helloworld-example` in the `runtime` component to set the project directory
+
+7. The `runtime` container will host the expressjs app which listens on port `3000`, define this port in the component by specifying an entry under 
+[`endpoints`](./devfile-schema#components-container-endpoints)
+    - Each endpoint has at least a `name` to identify them and the `targetPort` to specify the port number to forward
+
+        ```yaml {% title="Endpoints" %}
+        components:
+          - ...
+            container:
+              ...
+              endpoints:
```

Comment:
Would it be possible to turn this into a diff fence? It would help highlight what to edit in the devfile

Reply:
> Would it be possible to turn this into a diff fence? It would help highlight what to edit in the devfile

@mike-hoang Changed some of these to diff fences and added styles for highlighting, let me know if you have any feedback on these changes.

Reply:
Styling for this is now being taken care of in PR #88.